packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,210 @@
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+\=                                        return 'EQUALS';
+"*"                                       return 'MULT';
+\#                                        return 'BRKT';
+[\_]                                      return 'UNDERSCORE';
+"."                                       return 'DOT';
+"&"                                       return 'AMP';
+\-                                        return 'MINUS';
+[0-9]+                                    return 'NUM';
+\s+                                       /* skip */
+";"                                       return 'SEMI';
+<<EOF>>                                   return 'EOF';
+
+/lex
+
+%start start
+
+%% /* language grammar */
+
+start
+  : eol start
+  | directive start
+  | XYCHART chartConfig start
+  | XYCHART start
+  | document
+  ;
+
+chartConfig
+  : CHART_ORIENTATION                                           { yy.setOrientation($1); }
+  ;
+
+document
+  : /* empty */
+  | document statement
+  ;
+
+statement
+  : statement eol
+  | title text                                                  { yy.setDiagramTitle($text.text.trim()); }
+  | X_AXIS parseXAxis
+  | Y_AXIS parseYAxis
+  | LINE plotData                                               { yy.setLineData({text: '', type: 'text'}, $plotData); }
+  | LINE text plotData                                          { yy.setLineData($text, $plotData); }
+  | BAR plotData                                                { yy.setBarData({text: '', type: 'text'}, $plotData); }
+  | BAR text plotData                                           { yy.setBarData($text, $plotData); }
+  ;
+
+plotData
+  : SQUARE_BRACES_START commaSeperateNumber SQUARE_BRACES_END   { $$ = $commaSeperateNumber }
+  ;
+
+commaSeperateNumber
+  : NUMBER_WITH_DECIMAL commaSeperateNumberValues                 {
+                                                                    $commaSeperateNumberValues = $commaSeperateNumberValues || [];
+                                                                    $commaSeperateNumberValues.unshift(Number($NUMBER_WITH_DECIMAL));
+                                                                    $$ = $commaSeperateNumberValues
+                                                                  }
+  ;
+
+commaSeperateNumberValues
+  : COMMA NUMBER_WITH_DECIMAL commaSeperateNumberValues           {
+                                                                    $commaSeperateNumberValues = $commaSeperateNumberValues || [];
+                                                                    $commaSeperateNumberValues.unshift(Number($NUMBER_WITH_DECIMAL));
+                                                                    $$ = $commaSeperateNumberValues
+                                                                  }
+  |;
```

Comment:
There are similar comments to the other part:
1. _commaSep**a**rate**d**_
2. Can be united

```
commaSeparatedNumbers
  : NUMBER_WITH_DECIMAL COMMA commaSeparatedNumbers { $$ = [Number($NUMBER_WITH_DECIMAL), ...$commaSeparatedNumbers] }
  | NUMBER_WITH_DECIMAL  { $$ = [Number($NUMBER_WITH_DECIMAL)] }
  ;
```

Reply:
Done